### DISCUSSION TITLE
Overriding environment variables in .env file for recipe execution

### DISCUSSION THREAD

**Discussion by rohel01** [casey/just#1680] [Q&A] [answered]
I am using just to wrap a legacy build system, which is based on Makefiles. The current workflow is heavily reliant on developers sourcing configuration scripts (ex: Yocto) before running make. This is unpractical for at least two reasons:

Configuration scripts tend to interfere with each other in confusing ways
Tweaking the configuration requires restarting the user shells
This is why I am trying to move towards a just-based workflow with .env for shared configurations and explcit configuration script sourcing only in the relevant recipes.

Right now, I am trying to override the PATH from the .env file to expose basic utility tools (starting with make...) to every one. But my modifications are ignored because PATH is already defined in the environment.

Would the feature discussed in #642 help in my use case? Could you suggest another approach?

**Comment by runeimp** [comment, marked as answer]
If your recipe is run as a shebang script you can `source` the `.env` file which should take precedence in the rest of the recipe.

```sh
example:
	#!/usr/bin/env bash
	source .env
	# Do Stuff
```

**Reply by rohel01** [reply]
This still needs to be duplicated per recipe, but this is a very useful workaround. Thanks!
